src/webapp/reports/glass-data-submission/glass-data-submission-list/Filters.tsx

```diff
@@ -179,112 +178,135 @@ export const Filters: React.FC<DataSetsFiltersProps> = React.memo(props => {
-                        value={filter.to ?? null}
-                        minDate={filter.from}
-                        maxDate={new Date()}
-                        onChange={setEndDate}
+                ) : (
+                    <>
+                        <DatePickerStyled
+                            label="From"
+                            value={filterValues.from ?? null}
+                            maxDate={filterValues.to}
+                            onChange={setStartDate}
+                        />
+                        <DatePickerStyled
+                            label="To"
+                            value={filterValues.to ?? null}
+                            minDate={filterValues.from}
+                            maxDate={new Date()}
+                            onChange={setEndDate}
+                        />
+                    </>
+                )}
+
+                {dataSubmissionPeriod === "QUARTERLY" && (
+                    <DropdownStyled
+                        items={quarterItems}
+                        values={filterValues.quarters}
+                        onChange={setQuarters}
+                        label={i18n.t("Quarters")}
                     />
-                </>
-            )}
-
-            {dataSubmissionPeriod === "QUARTERLY" && (
-                <DropdownStyled
-                    items={quarterItems}
-                    values={filter.quarters}
-                    onChange={setQuarters}
-                    label={i18n.t("Quarters")}
-                />
-            )}
+                )}
+
+                {!isEARModule && (
+                    <SingleDropdownStyled
+                        items={completionStatusItems}
+                        value={fromBool(filterValues.completionStatus)}
+                        onChange={setCompletionStatus}
+                        label={i18n.t("Questionnaire completed")}
+                    />
+                )}
 
-            {!isEARModule && (
                 <SingleDropdownStyled
-                    items={completionStatusItems}
-                    value={fromBool(filter.completionStatus)}
-                    onChange={setCompletionStatus}
-                    label={i18n.t("Questionnaire completed")}
+                    items={isEARModule ? earSubmissionStatusItems : submissionStatusItems}
+                    value={filterValues.submissionStatus}
+                    onChange={setSubmissionStatus}
+                    label={i18n.t("Status")}
                 />
-            )}
-
-            <SingleDropdownStyled
-                items={isEARModule ? earSubmissionStatusItems : submissionStatusItems}
-                value={filter.submissionStatus}
-                onChange={setSubmissionStatus}
-                label={i18n.t("Status")}
-            />
-        </Container>
+            </Container>
+
+            <FilterButtonContainer>
+                <Button onClick={applyFilters} variant="contained" color="primary">
```

Comment:
Maybe you could add a disabled prop to enable the Apply filter button just when you can filter data (I think is when a module has been selected?), because was weird to click on apply data with for example just a year selected an nothing happends.